caikit/runtime/servicers/global_predict_servicer.py

```diff
@@ -202,6 +206,16 @@ def Predict(
                         request,
                         inference_signature,
                     )
+
+            # Get a unique ID from this request or make one up
+            if (
+                request_id := get_metadata(
+                    context, REQUEST_ID_HEADER_KEY, required=False
+                )
+            ) is None:
+                request_id = str(uuid.uuid4())
```

Comment:
suggestion: for cases where we are creating a local request_id, for ease of trackability, can we add a `local-` prefix to the `request_id` that way, when we are looking at the request_id, we know if this is something we got from downstream or we generated it locally

Reply:
Nice, this seems like a good idea

Reply:
Now that I think more about it, all of this `request_id` stuff should be stripped out. This was my original thought on how to do span propagation, but was not at all aligned with how otel does it.

One of the big follow ups I'd like to do from this PR is to create an _actual_ `RuntimeServerContextType` class that does the shimming between `grpc` and `fastapi` so that we don't need to keep writing optional dependency checks and type checks. If we do this, I think this would be a fine place to reinstate this `request_id` logic so that the id becomes available to any place downstream that wants to have it for logging purposes.

As it stands, though, the _only_ use of this `request_id` is to attach it to the trace which is useless because it will always have a unique ID for the `span` anyway.